test cases/common/139 override options/meson.build

```diff
@@ -0,0 +1,6 @@
+project('option override', 'c',
+  default_options : 'unity=true')
+
+executable('mustunity', 'one.c', 'two.c')
+executable('notunity', 'three.c', 'four.c',
+  override_options : ['unity=false'])
```

Comment:
Added werror test. You can't really override meson_options because these overrides are only used inside builtin targets and those do not look up project options.